### PR TITLE
GPIO Cleanup

### PR DESCRIPTION
**Pull request by kshxtij**
## Description
Clean up of gpio_manager.cpp, gpio_manager.hpp, gpio_counter.cpp, gpio_counter.hpp.

Removed all references to using.

### CODE REVIEW

File: src/sensors/gpio_manager.cpp [hunk 1]
```diff
@@ -6,19 +6,19 @@
...
     : Thread(log),
       sys_(utils::System::getSystem()),
-      data_(Data::getInstance())
+      data_(data::Data::getInstance())
 {
   // clear HPSSRs if default is high
-  for (int i = 0; i < data::Batteries::kNumHPBatteries; i++) {
-    hp_ssr_.push_back(new GPIO(sys_.config->sensors.hp_shutoff[i], utils::io::gpio::kOut));
+  for (std::size_t i; i < data::Batteries::kNumHPBatteries; ++i) {
+    hp_ssr_.emplace_back(new utils::io::GPIO(sys_.config->sensors.hp_shutoff[i], utils::io::gpio::kOut));
```
Comment (MiltFra) [inline src/sensors/gpio_manager.cpp:16]: this should be `hp_ssr_.emplace_back(sys.config->sensors.hp_shutoff[i], utils::io::gpio::kOut)`. Note that `new` performs a heap allocation and returns a pointer. However, we want to construct the object in the memory that has been allocated for the vector already.

**Comment by MiltFra** [comment]
Huh, seems like this still requires the copy constructor, didn't know that...

File: src/sensors/gpio_counter.hpp [hunk 2]
```diff
@@ -16,7 +16,7 @@ using utils::concurrent::Thread;
 
 namespace sensors {
 
-class GpioCounter : public GpioInterface, public Thread {  // interface.hpp
+class GpioCounter : public GpioInterface, public utils::concurrent::Thread {  // interface.hpp
  public:
   GpioCounter(utils::Logger &log, int pin);
```
Comment (suryanshmanocha) [inline src/sensors/gpio_counter.hpp:21]: const int

**Review by suryanshmanocha — CHANGES_REQUESTED** [review]

File: src/sensors/gpio_counter.hpp (hunk 2 above)
Comment (MiltFra) [reply src/sensors/gpio_counter.hpp:21]: Not even sure `int` is the right type here...

**Comment by MiltFra** [comment]
Closed in favour of #98.

**Pull request closed without merging**